test/500s.test.js

```diff
@@ -1,70 +1,75 @@
 'use strict'
 
-const t = require('tap')
-const test = t.test
+const { test } = require('node:test')
 const Fastify = require('..')
 const symbols = require('../lib/symbols.js')
 
-test('default 500', t => {
+test('default 500', async t => {
   t.plan(4)
 
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
+  t.after(() => fastify.close())
 
   fastify.get('/', function (req, reply) {
     reply.send(new Error('kaboom'))
   })
 
-  fastify.inject({
-    method: 'GET',
-    url: '/'
-  }, (err, res) => {
-    t.error(err)
-    t.equal(res.statusCode, 500)
-    t.equal(res.headers['content-type'], 'application/json; charset=utf-8')
-    t.same(JSON.parse(res.payload), {
-      error: 'Internal Server Error',
-      message: 'kaboom',
-      statusCode: 500
+  await new Promise((resolve, reject) => {
```

Comment:
Please don't wrap everything in new promises. The test function for this specific test should be like:

```js
(t, end) => {
	t.after(() => fastify.close())
	fastify.inject(params, (err, res) => {
		t.assert.errorIf(err)
		t.assert.ok(res)
		end()
	})
}
```

Reply:
ok, so I can use done().
what if there are 2 injects in one test? Should I still use done() in the second callback or wrap both in promises?

Reply:
If you really need to use promise, consider the below
```js
test('async', async (t) => {
  {
    const res = await fastify.inject(params)
    t.assert.equal(res.statusCode, 200)
  }
  {
    const res = await fastify.inject(params)
    t.assert.equal(res.statusCode, 200)
  }
})
```
It is more straight forward then wrapping inside promise.

Reply:
That's a good option as well. As I said, I tried to change as little as possible - so I kept working with callbacks.

If I'm going this way should I also wrap it with try/catch?

```
test('async', async (t) => {
  try {
    {
      const res = await fastify.inject(params)
      t.assert.equal(res.statusCode, 200)
    }
    {
       const res = await fastify.inject(params)
       t.assert.equal(res.statusCode, 200)
     }
   } catch (err) {
     t.assert.fail(err)
   }
})
```

Reply:
You don't need the `try-catch`, the `t.assert.errorIf` is used to throw when error.
That is the same for not asserting and let the test runner to catch the error itself.

Reply:
Will do. I thought using try/catch to get an assertion error still. But I guess it doesn't really matter.